lib/protoparser/opentelemetry/firehose/parser.go

```diff
@@ -32,8 +49,18 @@ func ProcessRequestBody(b []byte) ([]byte, error) {
 
 	var dst []byte
 	for _, r := range req.Records {
-		dst = append(dst, r.Data...)
+		for len(r.Data) > 0 {
+			messageLength, varIntLength := decodeVarint(r.Data)
```

Comment:
Why not using standard [encoding/binary.Uvarint](https://pkg.go.dev/encoding/binary#Uvarint) here?

Reply:
just updated